enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,608 @@
+annotation accepts multiple subnets, whereas GCP and Azure only permit a
+single subnet. For this reason, we made this API specific to AWS by adding
+the configuration in `AWSLoadBalancerParameters`.
+
+```go
+// AWSLoadBalancerParameters provides configuration settings that are
+// specific to AWS load balancers.
+// +union
+type AWSLoadBalancerParameters struct {
+  [...]
+
+  // subnets specifies the list of subnets for the load balancer to
+  // route traffic to. The values can be either a subnet ID or name.
+  // In order for the load balancer to be provisioned with subnets:
+  // * Each subnet must exist.
+  // * Each subnet must be from a different availability zone.
+  // * The load balancer service must be deleted.
+  //
+  // When omitted, the subnets will be auto-discovered per availability zone.
+  //
+  // +optional
+  // +kubebuilder:validation:MaxLength=256
+  // +openshift:enable:FeatureGate=IngressControllerLBSubnetsAWS
+  Subnets []AWSSubnetReference `json:"subnets,omitempty"`
+}
+
+// AWSSubnetReference is a reference to an AWS subnet. It can be either
+// be a subnet ID or name.
+// +kubebuilder:validation:MaxLength=256
+type AWSSubnetReference string
+```
+
+#### Effectuating Subnet Updates
+
+If a cluster admin updates the `Subnets` field on an existing IngressController,
+the Ingress Operator doesn't effectuate this change automatically. Instead, the
+Ingress Operator sets the `LoadBalancerProgressing` condition to `Status: True`
+provides a message indicating to the cluster admin that they must delete the service.
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: custom
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  domain: mycluster.com
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: External
+      providerParameters:
+        type: AWS
+        aws:
+          type: Classic
+          subnets:
+            - subnetA
+            - subnetB
+            - subnetC
+status:
+  availableReplicas: 2
+  conditions:
+  - message: Deployment has minimum availability.
+    reason: MinimumReplicasAvailable
+    status: "True"
+    type: Available
+  - message: "The IngressController subnets were changed from [\"\"] to [\"subnetA\", \"subnetB\", \"subnetC\"]. To effectuate this change, you must delete the service: `oc -n openshift-ingress delete svc/custom-default`; the service load-balancer will then be deprovisioned and a new one created. This will most likely cause the new load-balancer to have a different host name and IP address and cause disruption. To return to the previous state, you can revert the change to the IngressController: `oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"type\":\"LoadBalancerService\",\"loadBalancer\":{\"providerParameters\":{\"type\":\AWS\",\"aws\":{\"type\":\"Classic\",\"subnets\":[]}}}}}}'"
```

Comment:
Can you add line breaks so the message shows up in its entirety without scrolling?